.github/workflows/release.yaml

```diff
@@ -20,21 +23,124 @@ jobs:
+          
+            sam build --template-file "$TEMPLATE_FILE" --build-dir "$BUILD_DIR"
+            sam package --s3-bucket "$bucket_name" --template-file "$BUILD_DIR/template.yaml" --output-template-file "$BUILD_DIR/$TEMPLATE_FILE"  --region $region
+            if [[ $? -ne 0 ]]; then
+              echo "Error: Failed to package template $TEMPLATE_FILE"
+            fi  
+          
+            aws s3 cp "$BUILD_DIR/$TEMPLATE_FILE" "s3://$bucket_name/$TEMPLATE_FILE"
+            if [[ $? -ne 0 ]]; then
+              echo "Error: Failed to copy $TEMPLATE_FILE to S3 bucket $bucket_name"
+            fi
+          
+          done
+
+  deploy-lambda-code-parallelly-to-all-region:
+    needs: build-and-deploy
+    permissions:
+      id-token: write
+      contents: write
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        region:
+          - us-east-1
+          - us-east-2
+          - eu-west-1
+          - eu-west-2
+          - us-west-1
+          - us-west-2
+          - af-south-1
+          - ap-east-1
+          - ap-south-2
+          - ap-southeast-3
+          - ap-southeast-5
+          - ap-southeast-4
+          - ap-south-1
+          - ap-northeast-3
+          - ap-northeast-2
+          - ap-southeast-1
+          - ap-southeast-2
+          - ap-northeast-1
+          - ca-central-1
+          - ca-west-1
+          - eu-central-1
+          - eu-south-1
+          - eu-west-3
+          - eu-south-2
+          - eu-north-1
+          - eu-central-2
+          - me-south-1
+          - me-central-1
+          - il-central-1
+          - sa-east-1
+
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: log-forwarder-zip
+          path: ./build-artifacts
 
-      - name: Build SAM Application
-        run: sam build --template-file template.yaml --region us-east-2
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: ${{ secrets.AWS_S3_PUBLISH_ROLE }}
+          aws-region: us-east-2
 
-      - name: Package SAM Application
-        run: sam package --s3-bucket unified-lambda-cft-1 --output-template-file packaged.yaml --region us-east-2
-# should we include retries as part of uploading?
-      - name: Upload CloudFormation Template to S3
-        run: aws s3 cp packaged.yaml s3://unified-lambda-serverless-1/packaged.yaml
+      # should we include retries as part of uploading?
+      - name: Upload lambda code to S3 buckets in all AWS region
+        env:
+          bucket_prefix: new-relic-log-forwarder-folder
+          region: ${{ matrix.region }}
+        run: |
+          bucket_name="unified-logging-lambda-code-$region"
+          aws s3 cp "./build-artifacts/$log_forwarder_zip_file_name" "s3://$bucket_name/$bucket_prefix/" --region "$region"
+          echo "Deploying $log_forwarder_zip_file_name to region $REGION"
```

Comment:
1. You have some error logging for template file, should we add it here also?
2. How can we add retry?

Reply:
I'll add the error handling here as well

Reply:
We should be able to add retries here. I'll research more on this

Reply:
Added retry logic and error handling. Please review